lib/MetricsConsumer.js

```diff
@@ -164,14 +178,14 @@ class MetricsConsumer {
                 ops: 5,
                 bytes: 195,
                 extension: 'crr',
-                type: 'processed'
+                type: 'queued'
             }
         */
         // filter metric entries by service, i.e. 'crr', 'ingestion'
         if (this._id !== data.extension) {
             return done();
         }
-        const operationTypes = ['completed', 'failed', 'queued'];
+        const operationTypes = MetricsConsumer.getValidMetricTypes();
```

Comment:
I really like the extraction of this array and the idea of containing the logic. It seems to me this idea can be extended even further here to determine `isValidType`. For example, you could keep `getValidMetricTypes` and add another method `MetricsConsumer.isValidMetricType` which calls `getValidMetricTypes` internally and returns a boolean. 😸

Reply:
nice i like the idea of encapsulating both methods into one. So it will contain the list of valid types and return a boolean based on the type to check